### PR TITLE
Restart server and auto refresh browser on changes!

### PR DESCRIPTION
**Pull request by samarpanda**
## Restart server and auto refresh browser on changes!

Changing HTML files will refresh the browser automatically. Same goes for server side change i.e `index.js`
Added few dev dependencies for this feature `livereload` and `nodemon`.

``` shell
$ npm start

> falcor-express-demo@0.0.1 start /Users/samar/Projects/falcor-express-demo
> nodemon -w index.js & node watch.js

[nodemon] 1.8.1
[nodemon] to restart at any time, enter `rs`
[nodemon] watching: index.js
[nodemon] starting `node index.js`
```

### CODE REVIEW

**Comment by samarpanda** [comment]
cc: @tbranyen & @jameswomack

File: README.md [hunk 1]
```diff
@@ -8,9 +8,21 @@ This project is a demonstration of how to create a stateless Virtual JSON Graph
 clone repo
 npm install
 npm start
-open your browser and visit http://localhost:9090
+Automatically opens http://localhost:9090 on your default browser
```
Comment (tbranyen) [inline README.md]: Swap `on` for `in`.

File: README.md [hunk 2]
```diff
@@ -8,9 +8,21 @@ This project is a demonstration of how to create a stateless Virtual JSON Graph
...
 ```
 
+## Notes
+
+No manual refresh of browser on every change of html file.
```
Comment (tbranyen) [inline README.md]: Attempt at rephrasing:

"Changing HTML files will refresh the browser automatically."

**Comment by tbranyen** [comment]
Before I review further, any reason why you picked browser-sync instead of the ubiquitous LiveReload?

**Comment by samarpanda** [comment]
@tbranyen tried to make minimal / no change to current html file. Basically was trying to avoid injecting a js file for reloading although internally browser-sync does that. 

Hmm i think its a overhead here. May be i should replace browser-sync with LiveReload. Makes more sense as per this project. Do you want me take this?

**Comment by samarpanda** [comment]
@tbranyen replaced `browser-sync` with `livereload`. Hope this is fine now?

**Comment by tbranyen** [comment]
Can you squash down your commits into a single one? If you want I'd be happy to touch up some of the English and syntax formatting. Just add me to your repo and I'll add some commits in.

**Comment by samarpanda** [comment]
@tbranyen I have added you to my forked repo. :+1:

**Comment by tbranyen** [comment]
Awesome, I'll fix up the commits so that they're all squashed down and remove that recently added merge commit.

**Comment by samarpanda** [comment]
Cool :+1:

**Comment by samarpanda** [comment]
@tbranyen Let me know if i can help any way? Curious to see this merged to master :+1:

**Comment by tbranyen** [comment]
Yup, been busy lately, will get to it today :)

**Comment by samarpanda** [comment]
Fantastic :)

**Comment by tbranyen** [comment]
Welp, totally lied. Looks like today means today :)

**Comment by tbranyen** [comment]
@samarpanda could you do me a favor and review the changes in my commit? Let me know if it all looks okay.

File: README.md [hunk 3]
```diff
@@ -4,13 +4,27 @@ This project is a demonstration of how to create a stateless Virtual JSON Graph
...
 npm start
-open your browser and visit http://localhost:9090
 ```
 
+Open your browser and visit: [http://localhost:9090](http://localhost:9090)
+
+## Notes
+
+Changing the `index.html` file will refresh the browser automatically using
+[LiveReload][2]. Changing the server `index.js` file will reload the server
```
Comment (samarpanda) [inline README.md]: `restart` could be a more appropriate word here than `reload` as per server context. What do you say?

File: index.html [hunk 4]
```diff
@@ -72,6 +72,8 @@
         // model.get('genrelist.myList.titles.length').then(jlog, jerror)
 
     </script>
+
+    <script src="http://localhost:35729/livereload.js?snipver=1"></script>
```
Comment (samarpanda) [inline index.html:76]: :+1:

File: package.json [hunk 5]
```diff
@@ -27,6 +27,9 @@
     "falcor-express": "^0.1.2"
   },
   "devDependencies": {
-    "eslint": "^1.7.1"
+    "eslint": "^1.7.1",
+    "browser-sync": "^2.8.2",
```
Comment (samarpanda) [inline package.json]: We are not using `browser-sync` anymore. Better to remove this from here. I think i removed this in last commit.

**Comment by samarpanda** [comment]
I have pushed a commit addressing my comments. Not sure if you want me squash?
@tbranyen looks good :+1:

File: README.md (hunk 3 above)
Comment (tbranyen) [inline README.md]: I agree

Comment (samarpanda) [inline README.md]: cool

**Comment by tbranyen** [comment]
No, lets not squash I think it looks good!

**Comment by jameswomack** [comment]
:clap:

**Comment by jameswomack** [comment]
:ship: :it:

**Comment by samarpanda** [comment]
Cool, lets :ship: :shipit: :rocket:

**Pull request merged**